src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -166,16 +166,42 @@ public function xmlHttpRequest(string $method, string $uri, array $parameters =
      */
     public function jsonRequest(string $method, string $uri, array $parameters = [], array $server = [], bool $changeHistory = true): Crawler
     {
-        $content = json_encode($parameters);
+        // Based on https://github.com/symfony/symfony/blob/v5.2.0/src/Symfony/Component/HttpFoundation/Request.php#L388-L404
+        // the logic in symfony/http-foundation we convert parameters to json content.
+        switch (strtoupper($method)) {
+            case 'POST':
+            case 'PUT':
+            case 'DELETE':
+            case 'PATCH':
+                $content = json_encode($parameters);
+                $query = [];
+                break;
+            default:
+                $content = null;
+                $query = $parameters;
```

Comment:
wait, I'm not sure anymore that this makes sense:
$paraemters doesn't go through the processing of `json_encode()`
that means objects (that do or don't implement JsonSerializable) are going to be passed differently when using GET vs other verbs.

Reply:
Yeah based on the method we set it a request body or not. This is based on the linked logic which symfony currently use here: https://github.com/symfony/symfony/blob/v5.2.0/src/Symfony/Component/HttpFoundation/Request.php#L388-L404
It does not make sense to convert query parameters to JSON as nobody will converted it back. Which is in the FosRestBundle and ApiPlatform converted into the `$request->request` attributes by the [BodyListener](https://github.com/FriendsOfSymfony/FOSRestBundle/blob/3.x/EventListener/BodyListener.php).